### PR TITLE
RIA-3308: Changes to trigger HomeOffice notifications for request-evidence-bundle event

### PR DESCRIPTION
**Pull request by jkumar-moj**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3308

### Change description ###
Changes to trigger HomeOffice notifications for request-evidence-bundle event

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/functionalTest/resources/scenarios/RIA-3308-home-office-notifications-for-request_respondent_evidence.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      "state": "appealSubmitted",
+      "caseData": {
+        "template": "minimal-appeal-submitted.json",
+        "replacements": {
+          "homeOfficeReferenceNumber": "1111-2222-3333-4444",
+          "appealSubmissionDate": "2020-09-22",
+          "sendDirectionExplanation": "A notice of appeal has been lodged against this decision.\n\nYou must now upload all documents to the Tribunal. The Tribunal will make them accessible to the other party. You have 14 days to supply the documents.\n\nYou must include:\n- the notice of decision\n- any other document provided to the appellant giving reasons for that decision\n- any statements of evidence\n- the application form\n- any record of interview with the appellant in relation to the decision being appealed\n- any other unpublished documents on which you rely\n- the notice of any other appealable decision made in relation to the appellant",
+          "sendDirectionParties": "respondent",
+          "sendDirectionDateDue": "{$TODAY+14}",
+          "isHomeOfficeIntegrationEnabled": "Yes"
```
Comment (divisathyan) [inline src/functionalTest/resources/scenarios/RIA-3308-home-office-notifications-for-request_respondent_evidence.json:18]: The test will run if featureFlag is enabled. Do we have to set the variable again?

**Review by divisathyan — CHANGES_REQUESTED** [review]
Please see comment on isHomeOfficeIntegrationEnabled flag

File: src/functionalTest/resources/scenarios/RIA-3308-home-office-notifications-for-request_respondent_evidence.json (hunk 1 above)
Comment (jkumar-moj) [reply src/functionalTest/resources/scenarios/RIA-3308-home-office-notifications-for-request_respondent_evidence.json:18]: We use the the existing featureFlag for vaildation now. This is added here to depict that at this stage this flag should be set. There will be a dedicated flag added for notifications later.

Comment (jkumar-moj) [reply src/functionalTest/resources/scenarios/RIA-3308-home-office-notifications-for-request_respondent_evidence.json]: It's been removed.

**Review by divisathyan — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return isHomeOfficeIntegrationEnabled
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java:36]: Let's try to do this in Launch Darkly. It is recommended by PE now. I think we can still have HO validation flag and F&P ad they epics are ready but for new ones we should stick to Launch Darkly

**Review by makowalski — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java (hunk 2 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java]: Launch Darkly feature flag added

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by bradsorour — APPROVED** [review]
LGTM

**Pull request merged**